### PR TITLE
New article: Catching dreams

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

Here is my second article for Perl advent calendar 🎅🏼 

I probably need some help to arrange a bit the story in addition to usual edits!

### CODE REVIEW

**Comment by s1037989** [comment]
I haven't started reading your article yet, but I think it would be easiest -- like last time -- for the article to be merged and for me to make a PR on it rather than to review yours with comments throughout.  So I'll reviews and approve yours first.

**Review by s1037989 — APPROVED** [review]
I'll review via PR as opposed to via comments.

**Comment by s1037989** [comment]
I guess my approval isn't good enough.  :)

**Comment by s1037989** [comment]
Dang it...  I'll never get used to the fact that I need to go to the forked repo...  PR submitted.

**Comment by thibaultduponchelle** [comment]
Merged! Thank you a lot for the review!

**Comment by thibaultduponchelle** [comment]
@adamcrussell Would you be available to review this other article? Please 🙏🏼

File: 2022/incoming/catching-dreams.pod [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Let's install L<Mango>:
+
+	$ sudo cpanm Mango
+
+Then they can access records with Perl methods:
+
+=begin perl
+	
+	use Mango;
+	use feature state;
```
Comment (oalders) [inline 2022/incoming/catching-dreams.pod:82]:
```suggestion
	use feature qw( state );
```

Looks like this is copy/paste from a broken SYNOPSIS: https://github.com/oliwer/mango/commit/260788cb5e3ac1c0f22f449eda61d1b941ec577f

File: 2022/incoming/catching-dreams.pod [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+	Transformer for Billy in Houston
+	bicycle for Tom in Atlanta
+	ball for Ismail in Atlanta
+	Barbie for Lisa in Berlin
+
+
+=head2 Give me dreams from Atlanta
+
+When it comes time to doing deliveries, Santa is interested in having records from the same area to optimize the logistics of his delivery schedule.
```
Comment (oalders) [inline 2022/incoming/catching-dreams.pod:104]:
```suggestion
When it comes time for making deliveries, Santa is interested in having records from the same area to optimize the logistics of his delivery schedule. 
```

File: 2022/incoming/catching-dreams.pod [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+=head2 Give me dreams from Atlanta
+
+When it comes time to doing deliveries, Santa is interested in having records from the same area to optimize the logistics of his delivery schedule. 
+
+With L<Mango>, it's just a matter of adding a "clause" in the `find()` function:
+
+=begin perl
+
+	my $docs = mango->db('christmas2022')->collection('gifts')->find({where => "Atlanta"});
```
Comment (oalders) [inline 2022/incoming/catching-dreams.pod:110]:
```suggestion
	my $docs = mango->db('christmas2022')->collection('gifts')->find({where => 'Atlanta'});
```

I think it's helpful to have the quotes be consistent.

File: 2022/incoming/catching-dreams.pod [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+=end perl
+
+=head2 Conclusion
+
+With the elves copious year-round notetaking of childrens' dreams, Santa can count on the elves to not miss any of the childrens' wishes for Christmas.
```
Comment (oalders) [inline 2022/incoming/catching-dreams.pod:116]:
```suggestion
With the elves' copious year-round notetaking of children's dreams, Santa can count on the elves not to miss any of the children's wishes for Christmas. 
```

File: 2022/incoming/catching-dreams.pod [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+Let's install L<Mango>:
+
+	$ sudo cpanm Mango
+
+Then they can access records with Perl methods:
+
+=begin perl
+	
+	use Mango;
```
Comment (oalders) [inline 2022/incoming/catching-dreams.pod:81]:
```suggestion
	use Mango ();
```

**Review by oalders — CHANGES_REQUESTED** [review]
I love it! Just a few suggestions. 🧝🏻 🥭

File: 2022/incoming/catching-dreams.pod [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+They bring a pen and a notebook, and they listen to childrens' dreams and take notes.
+
+=head2 Gathering notebooks
+
+Elves capture notes on childrens' dreams all year long and, hence, approaching Christmas there are plenty of notebooks to put together and add to other wishes.
+
+One of the elves had the idea to use L<MongoDB|https://www.mongodb.com/> to store data.
+
+Santa complained MongoDB was not Perl friendly (e.g. struggles with drivers...) which had in its history taken a controversial license... But the elves are not disciplined and decided to go with it anyway.
```
[the remark this answers is not in the document]
Comment (thibaultduponchelle) [reply 2022/incoming/catching-dreams.pod]: Re-licensing in 2018 to SSPL. It can be considered as controversial.

**Comment by thibaultduponchelle** [comment]
@oalders It's ready to merge now (or some new edits 😄)

**Review by oalders — APPROVED** [review]

**Comment by oalders** [comment]
@thibaultduponchelle can you squash it down to one commit?

**Comment by thibaultduponchelle** [comment]
> @thibaultduponchelle can you squash it down to one commit?

Ok 👍will do it

@s1037989 Can you please approve?

**Comment by thibaultduponchelle** [comment]
Done! 💯

**Pull request merged**